### PR TITLE
stm32: Update openocd.cfg to support JLink debug adapter

### PR DESCRIPTION
**Pull request by qianfan-Zhao**
Commit af05ff6c introduct a way flashing and debuging using JLink tool,
that's using one more param '-DSTLINK_FW' when using CMake generates
the build system.

And the previous way can't change the debug tool unless you clean all
CMake files and regenerated. This patch using a environment variable
to select which debug tool we used, so you can select another tool
at anytimes, just change the envrionment variable.

Signed-off-by: qianfan Zhao <qianfanguijin@163.com>

### CODE REVIEW

**Comment by qianfan-Zhao** [comment]
@erwango Please also check it.

I tested this pr on nucleo-f070rb and nucleo-f207zg, so just only changed this two board.

**Comment by erwango** [comment]
I don't know if using env variable is acceptable or not (If it fits all environments).
Adding @mbolivar in the review

**Comment by qianfan-Zhao** [comment]
@erwango
this pr is copy ateml sam soc's idea, That's already merged in master. please see boards/sam_e70_xplained

File: doc/tools/stlink-v21.rst [hunk 1]
```diff
@@ -59,15 +59,14 @@ Host Tools and Firmware
 Download and install the `Segger J-Link Software and Documentation Pack`_ to
 get the J-Link GDB server for your host computer.
 
+Open new terminal and export ``OPENOCD_INTERFACE=jlink`` environment variable
```
Comment (dbkinder) [inline doc/tools/stlink-v21.rst]: Change to:

    To make OpenOCD use the J-Link tools, open a new terminal and use::

        export OPENOCD_INTERFACE=jlink

    before flashing and debugging.

**Review by dbkinder — CHANGES_REQUESTED** [review]

**Comment by qianfan-Zhao** [comment]
recheck

**Review by mbolivar — CHANGES_REQUESTED** [review]
Thanks for the pull request!

Indeed as @erwango said this does not seem particularly Windows-friendly; we are trying to get away for environment variables. Just because there's existing code in the tree which uses them doesn't mean we should keep adding more.

Why not add an argument to the openocd west runner? https://docs.zephyrproject.org/latest/west/flash-debug.html

That would be better and more future proof. Am I misunderstanding something that would make this possible?

Please note that pull requests must be submitted to upstream west for now if you go that route. https://github.com/zephyrproject-rtos/west

**Comment by qianfan-Zhao** [comment]
@mbolivar Thanks yours guide.
Actually I haven't reading zephyr's doc carefully, also I am not a python developer don't know how west works, I am trying learn python to figure out how it works.

I will try 'west' based on yours guide after holiday.

thanks!

**Review by dbkinder — APPROVED** [review]
+1 for doc changes

**Comment by mbolivar** [comment]
> I will try 'west' based on yours guide after holiday.

Sounds great! Looking forward to your contribution.

**Comment by mbolivar** [comment]
@qianfan-Zhao by the way, please feel free to submit a feature request to the west repository for this by filing an issue if you find that you don't have time to work on this after your holiday.

**Pull request closed without merging**